src/plotting.jl

```diff
@@ -0,0 +1,5 @@
+# Note: recipes seem to need to use short-form functions
```

Comment:
They don't need short form functions, they work with any function definition syntax.

Reply:
Doesn't seem to  AFAICT.

If I change it to:
```
@recipe function f(::Type{T}, val::T) where {T<:ZonedDateTime}
    return (
        zdt -> Dates.value(DateTime(zdt, UTC)),
        dt -> string(DateTime(Dates.UTM(dt))),
    )
end
```

my tests fail with:

```
ZonedDateTime plot recipe: Error During Test at /Users/oxinabox/JuliaEnvs/TimeZones.jl/test/plotting.jl:1
  Got exception outside of a @test
  type #31#33 has no field args
  Stacktrace:
   [1] getproperty(::Function, ::Symbol) at ./Base.jl:20
   [2] _apply_type_recipe(::Dict{Symbol,Any}, ::StepRange{ZonedDateTime,Hour}) at /Users/oxinabox/.julia/packages/Plots/rNwM4/src/series.jl:183
   [3] macro expansion at /Users/oxinabox/.julia/packages/Plots/rNwM4/src/series.jl:228 [inlined]
   [4] apply_recipe(::Dict{Symbol,Any}, ::StepRange{ZonedDateTime,Hour}, ::UnitRange{Int64}) at /Users/oxinabox/.julia/packages/RecipesBase/G4s6f/src/RecipesBase.jl:279
   [5] _process_userrecipes(::Plots.Plot{Plots.UnicodePlotsBackend}, ::Dict{Symbol,Any}, ::Tuple{StepRange{ZonedDateTime,Hour},UnitRange{Int64}}) at /Users/oxinabox/.julia/packages/Plots/rNwM4/src/pipeline.jl:85
   [6] _plot!(::Plots.Plot{Plots.UnicodePlotsBackend}, ::Dict{Symbol,Any}, ::Tuple{StepRange{ZonedDateTime,Hour},UnitRange{Int64}}) at /Users/oxinabox/.julia/packages/Plots/rNwM4/src/plot.jl:178
   [7] #plot#138(::Base.Iterators.Pairs{Symbol,Symbol,Tuple{Symbol},NamedTuple{(:seriestype,),Tuple{Symbol}}}, ::typeof(plot), ::StepRange{ZonedDateTime,Hour}, ::Vararg{Any,N} where N) at /Users/oxinabox/.julia/packages/Plots/rNwM4/src/plot.jl:57
   [8] (::RecipesBase.var"#kw##plot")(::NamedTuple{(:seriestype,),Tuple{Symbol}}, ::typeof(plot), ::StepRange{ZonedDateTime,Hour}, ::UnitRange{Int64}) at ./none:0
   [9] #scatter#402(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(scatter), ::StepRange{ZonedDateTime,Hour}, ::Vararg{Any,N} where N) at /Users/oxinabox/.julia/packages/RecipesBase/G4s6f/src/RecipesBase.jl:393
   [10] scatter(::StepRange{ZonedDateTime,Hour}, ::Vararg{Any,N} where N) at /Users/oxinabox/.julia/packages/RecipesBase/G4s6f/src/RecipesBase.jl:393
   [11] top-level scope at /Users/oxinabox/JuliaEnvs/TimeZones.jl/test/plotting.jl:6
   [12] top-level scope at /usr/local/src/julia/julia-1.3/usr/share/julia/stdlib/v1.3/Test/src/Test.jl:1107
   [13] top-level scope at /Users/oxinabox/JuliaEnvs/TimeZones.jl/test/plotting.jl:2
   [14] include at ./boot.jl:328 [inlined]
   [15] include_relative(::Module, ::String) at ./loading.jl:1105
   [16] include(::Module, ::String) at ./Base.jl:31
   [17] include(::String) at ./client.jl:424
   [18] top-level scope at /Users/oxinabox/JuliaEnvs/TimeZones.jl/test/runtests.jl:73
```

Reply:
what if you do:
```julia
@recipe function f(::Type{<: ZonedDateTime}, val::ZonedDateTime)
```
instead?

Reply:
same error.